assembly/src/assembler/instruction/u32_ops.rs

```diff
@@ -459,117 +348,57 @@ fn handle_division(
 /// Mutate the first two elements of the stack from `[b, a, ..]` into `[2^b, a, ..]`, with `b`
 /// either as a provided immediate value, or as an element that already exists in the stack.
 ///
-/// If the used mode is `checked`, the function will assert that both `[b, a]` are valid `u32`.
-/// This function is equivalent to a bit shift operation, so the exponent shouldn't cause a number
-/// to be greater than `u32::MAX`; therefore, the maximum valid value must be `31`, as defined in
-/// the helper constants.
-///
-/// This function supports only checked and unchecked modes; if some other mode is provided, it
-/// will panic.
+/// This function supports only unchecked mode; if some other mode is provided, it will panic.
 fn prepare_bitwise<const MAX_VALUE: u8>(
     span: &mut SpanBuilder,
     imm: Option<u8>,
-    op_mode: U32OpMode,
-    final_ops: [Operation; 2],
 ) -> Result<Option<CodeBlock>, AssemblyError> {
```

Comment:
We can probably return just `Result<(), AssemblyError>` from here and in the function use `push_op` instead of `add_op`.

Also, we probably don't need to mention "unchecked" mode in the comment.